ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -48,42 +57,162 @@ static int gcp_remove(WT_FILE_SYSTEM *, WT_SESSION *, const char *, uint32_t)
   __attribute__((__unused__));
 static int gcp_rename(WT_FILE_SYSTEM *, WT_SESSION *, const char *, const char *, uint32_t)
   __attribute__((__unused__));
-static int gcp_file_size(WT_FILE_HANDLE *, WT_SESSION *, wt_off_t *) __attribute__((__unused__));
+static int gcp_file_size(WT_FILE_SYSTEM *, WT_SESSION *, const char *, wt_off_t *)
+  __attribute__((__unused__));
+static int gcp_object_list(
+  WT_FILE_SYSTEM *, WT_SESSION *, const char *, const char *, char ***, uint32_t *);
 static int gcp_object_list_add(const gcp_store &, char ***, const std::vector<std::string> &,
   const uint32_t) __attribute__((__unused__));
 static int gcp_object_list_single(WT_FILE_SYSTEM *, WT_SESSION *, const char *, const char *,
   char ***, uint32_t *) __attribute__((__unused__));
 static int gcp_object_list_free(WT_FILE_SYSTEM *, WT_SESSION *, char **, uint32_t)
   __attribute__((__unused__));
+static int gcp_terminate(WT_STORAGE_SOURCE *, WT_SESSION *) __attribute__((__unused__));
 
 static int
-gcp_customize_file_system(WT_STORAGE_SOURCE *store, WT_SESSION *session, const char *bucket,
-  const char *auth_token, const char *config, WT_FILE_SYSTEM **file_system)
+gcp_customize_file_system(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
+  const char *bucket, const char *auth_token, const char *config, WT_FILE_SYSTEM **file_system)
 {
-    WT_UNUSED(store);
-    WT_UNUSED(session);
-    WT_UNUSED(bucket);
-    WT_UNUSED(auth_token);
-    WT_UNUSED(config);
-    WT_UNUSED(file_system);
+    // Check if bucket name is given
+    if (bucket == nullptr || strlen(bucket) == 0) {
+        std::cerr << "gcp_customize_file_system: bucket not specified." << std::endl;
+        return EINVAL;
+    }
+
+    // Fail if there is no authentication provided.
+    if (auth_token == nullptr || strlen(auth_token) == 0) {
+        std::cerr << "gcp_customize_file_system: auth_token not specified." << std::endl;
+        return EINVAL;
+    }
+
+    int delimiter = std::string(auth_token).find(".json");
+    if (delimiter == std::string::npos || delimiter == 0) {
+        std::cerr << "gcp_customize_file_system: improper auth_token: " + std::string(auth_token) +
```

Comment:
Let's try to avoid casual grammar in the error messaging (such as stringing thoughts together by commas). 

I think the error message is misleading. It looks like you are checking that the input parameter has the expected file extension. Perhaps update the message to "the specified authentication file must have a .json extension."? 

I suggest using `std::filesystem::path::extension` rather than `find`, if you are in fact checking for a file extension.